File: 57. Insert Interval.py
class Solution(object):
    def insert(self, intervals, newInterval):
        """
        :type intervals: List[List[int]]
        :type newInterval: List[int]
        :rtype: List[List[int]]
        """
        if intervals == []:
            return [newInterval]
        newinterval = []
        dowork = []
        stop = 0
        for i in range(len(intervals)):
            if intervals[i][1]<newInterval[0]:
                newinterval.append(intervals[i])
            elif newInterval[1]>=intervals[i][0]:
                dowork.append(intervals[i])
            else:
                break
            stop += 1
        if dowork != []:
            newinterval.append([min(dowork[0][0],newInterval[0]),max(dowork[-1][-1],newInterval[1])])
        else:
            newinterval.append(newInterval)

        for j in range(stop,len(intervals)):
            newinterval.append(intervals[j])
        return newinterval

test = Solution()
print(test.insert([[1,2]],[6,8]))

File: CPP/57. Insert Interval.cpp
using namespace std;
#include <vector>

class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        vector<vector<int>> ret;
        bool inserted = false;
        for (int i = 0; i < intervals.size(); i++){
            vector<int> curr = intervals.at(i);
            if (inserted){
                ret.push_back(curr);
                continue;
            }
            // no overlapping
            if (curr.at(1) < newInterval.at(0)){
                ret.push_back(curr);
            } else if (curr.at(0) > newInterval.at(1)) {
                ret.push_back(newInterval);
                ret.push_back(curr);
                inserted = true;
            } else {
                newInterval.at(0) = min(newInterval.at(0), curr.at(0));
                newInterval.at(1) = max(newInterval.at(1), curr.at(1));
            }
        }
        if (!inserted){
            ret.push_back(newInterval);
        }
        return ret;
    }
};